lib/puppetserver/ca/action/generate.rb

```diff
@@ -140,7 +144,11 @@ def run(input)
           # Generate and save certs and associated keys
           if input['ca-client']
             # Refused to generate certs offfline if the CA service is running
-            return 1 if HttpClient.check_server_online(puppet.settings, @logger)
+            if HttpClient.check_server_online(puppet.settings, @logger, input['force'])
+              return 1
+            else
+              @logger.inform("Continuing with certificate signing.") if input['force']
+            end
             all_passed = generate_authorized_certs(certnames, alt_names, puppet.settings, signer.digest)
           else
```

Comment:
Perhaps I could add a logging statement in this `else` block  that says "--force flag is ineffective without --ca-client flag, proceed with signing certificate while Puppet Server is online."

Reply:
Lemme think out loud about this for a sec.. 

It looks like we only `check_server_online` when `--ca-client` is used, i.e. the `generate` action can be run while the server is online, except for when `--ca-client` is used. 
So in that case, I can't think of what someone would expect from using the `--force` flag without `--ca-client`.. so maybe it is best to leave it as-is, just doing nothing.. 

Does that make sense @Magisus @justinstoller ?

Reply:
Yeah agreed, if we were going to warn about the misused flag, I would do it during args parsing at the very beginning, and not down here.

Reply:
Warning or erroring early during argument validation (e.g. "Error: --force must be used with --ca-client") makes sense to me too.